src/common/buffer.cc

```diff
@@ -1582,6 +1587,12 @@ static simple_spinlock_t buffer_debug_lock = SIMPLE_SPINLOCK_INITIALIZER;
       push_back(bp);
   }
 
+  void buffer::list::append(ptr&& bp)
+  {
+    if (bp.length())
+      push_back(bp);
```

Comment:
bp is an lvalue in the body of the function, so the call to push_back() needs to cast it again: `push_back(std::move(bp));`